### PR TITLE
better logging of errors

### PR DESCRIPTION
**Pull request by MaxVerevkin**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -148,14 +148,22 @@ fn main() -> Result<()> {
                 }
                 Err(error) => {
                     error!("{error}");
+                    error
```
Comment (pierrechevalier83) [inline src/main.rs:151]: Ah! Thanks for the tip. I haven't used anyhow before, so I wasn't sure how to retrieve the rest of the context from the error. That's nice.

File: src/window_manager.rs [hunk 2]
```diff
@@ -137,21 +128,13 @@ impl WindowManager {
     pub fn get_windows_in_each_workspace(&mut self) -> Result<BTreeMap<String, Vec<Window>>> {
         self.connection
             .get_tree()
```
Comment (pierrechevalier83) [inline src/window_manager.rs:130]: This one gets removed but replaced by nothing as far as I can see.

File: src/window_manager.rs [hunk 3]
```diff
@@ -137,21 +128,13 @@ impl WindowManager {
...
-                log::error!("{e}");
-                e
-            })
             .context("get_tree() failed")?
             .workspaces_in_node()
     }
 
     pub fn rename_workspace(&mut self, old: &str, new: &str) -> Result<()> {
         self.connection
             .run_command(&format!("rename workspace \"{old}\" to \"{new}\"",))
```
Comment (pierrechevalier83) [inline src/window_manager.rs:137]: This one gets removed but replaced by nothing as far as I can see.

**Review by pierrechevalier83 — APPROVED** [review]

File: src/window_manager.rs [hunk 4]
```diff
@@ -14,7 +14,7 @@ trait NodeExt {
 
 impl NodeExt for Node {
     fn is_workspace(&self) -> bool {
-        self.node_type == NodeType::Workspace
+        self.name.as_deref() != Some("__i3_scratch") && self.node_type == NodeType::Workspace
```
Comment (pierrechevalier83) [inline src/window_manager.rs:17]: Could you add a comment here explaining what this `i3_scratch` is? (or in the commit message if you prefer)

Comment (MaxVerevkin) [reply src/window_manager.rs:17]: To be honest IDK what this is but `rename_workspace()` errors when we try to rename it. We just weren't catching these errors previously.

Comment (MaxVerevkin) [reply src/window_manager.rs:17]: I'll add a comment explaining this

Comment (pierrechevalier83) [reply src/window_manager.rs:17]: And is that on i3 only?

Comment (MaxVerevkin) [reply src/window_manager.rs:18]: I use `sway`

Comment (pierrechevalier83) [reply src/window_manager.rs:18]: Some context from looking around a little bit:
https://github.com/swaywm/sway/blob/b8995ced8fb0a4c48300190dfd610f7420aca795/sway/ipc-json.c#L370
https://www.rockyourcode.com/til-about-the-i3-scratchpad/

Comment (MaxVerevkin) [reply src/window_manager.rs:18]: Thanks for pointing out!

**Review by pierrechevalier83 — APPROVED** [review]
Thanks for the PR @MaxVerevkin , looks great.

**Pull request merged**